static/app/views/monitors/components/checkInTimeline.tsx

```diff
@@ -0,0 +1,63 @@
+import styled from '@emotion/styled';
+
+import {Resizeable} from 'sentry/components/replays/resizeable';
+import {space} from 'sentry/styles/space';
+import {CheckIn, CheckInStatus} from 'sentry/views/monitors/types';
+
+interface Props {
+  checkins: CheckIn[];
+  end: Date;
+  start: Date;
+}
+
+function getCheckInPosition(checkDate: string, timelineStart: Date, msPerPixel: number) {
+  const elapsedSinceStart = new Date(checkDate).getTime() - timelineStart.getTime();
+  return elapsedSinceStart / msPerPixel;
+}
+
+export function CheckInTimeline({checkins, start, end}: Props) {
+  return (
+    <Resizeable>
```

Comment:
In the future we may want to make this optional since when we have a bunch of these we'll just want to wrap them in one.

Reply:
that's fair, i was considering that but it felt weird to accept `width` as a prop to this component, but would that be the future here?

Reply:
Yeah I think so, and without it it will use the resizing hook

(Probably means the hook will also need a option to turn it off so that it doesn't run if we pass a width)